third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -148,35 +147,21 @@ impl Interpreter {
         }
 
         let mut current_frame = self
-            .make_new_frame(
-                gas_meter,
-                entrypoint_resolver.loader(),
-                function,
-                ty_args,
-                locals,
-            )
+            .make_new_frame(gas_meter, loader, function, locals)
             .map_err(|err| self.set_location(err))?;
 
         // Access control for the new frame.
         self.access_control
-            .enter_function(&current_frame, current_frame.function.as_ref())
+            .enter_function(&current_frame, current_frame.function.definition())
             .map_err(|e| self.set_location(e))?;
 
         loop {
-            // Select which resolver to use: if we are executing in the context of the entrypoint
-            // function, we use the one provided by the caller.
-            let resolver = if self.call_stack.is_empty() {
-                BorrowedOrOwnedResolver::Borrowed(entrypoint_resolver)
-            } else {
-                let new_resolver = Resolver::new_for_function_in_module(
-                    current_frame.function.as_ref(),
-                    entrypoint_resolver.loader(),
-                    entrypoint_resolver.module_store(),
-                    entrypoint_resolver.module_storage(),
-                )
-                .map_err(|e| e.finish(Location::Undefined))?;
-                BorrowedOrOwnedResolver::Owned(new_resolver)
-            };
+            let resolver = Resolver::new_for_loaded_function(
```

Comment:
Now setting up the resolver from the current loaded function is trivial, as we store it in the frame cc @runtian-zhou